bindings/node/lib/common.js

```diff
@@ -46,6 +46,19 @@ class MongoCryptError extends Error {
   }
 }
 
+function maybeCallback(promiseFn, callback) {
+  const promise = promiseFn();
+  if (callback == null) {
+    return promise;
+  }
+
+  promise.then(
+    result => callback(undefined, result),
+    error => callback(error)
```

Comment:
Note that this will make exceptions from the callback result in an unhandled rejection rather than an uncaught exception; that’s a visible behavioral change. You may want to do something like this instead:

```suggestion
    result => process.nextTick(callback, undefined, result),
    error => process.nextTick(callback, error)
```

Reply:
Neal and I looked at this for a while and, while we believe you're right, we actually couldn't reproduce the issue.

```javascript
async function wait() {
  await setTimeout(1000);
}
maybeCallback(() => wait(), () => { throw new Error("crashed process") });
```

Doesn't result in an UnhandledPromiseRejection

Reply:
Are you _sure_? Running this code (as `node test.js` with Node.js 16 through 19 at least) results in the unhandled rejection handler being called:

```js

function maybeCallback(promiseFn, callback) {
  const promise = promiseFn();
  if (callback == null) {
    return promise;
  }

  promise.then(
    result => callback(undefined, result),
    error => callback(error)
  );
  return;
}

async function wait() {
  await setTimeout(1000);
}

process.on('unhandledRejection', (...args) => console.log('unhandledRejection', args));

maybeCallback(() => wait(), () => { throw new Error("crashed process") });
```

Reply:
ah, we were looking for the unhandledRejection console log message, ty! We should make this improvement here, we'll want to file a follow up to replicate it elsewhere we've wrapped callbacks

Reply:
Yup, you're right.  For some reason, the unhandled promise isn't being printed as an unhandled promise (we should have listened for unhandled promises).  

We filed a follow up ticket for the driver and legacy driver: https://jira.mongodb.org/browse/NODE-4996

Reply:
The default mode for unhandled rejections was changed to throwing in Node.js 15, which is why I think you weren’t seeing this printed as a warning: https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V15.md#throw-on-unhandled-rejections---33021